clients/drcachesim/simulator/cache_simulator.cpp

```diff
@@ -277,40 +278,51 @@ cache_simulator_t::process_memref(const memref_t &memref)
         return false;
     }
 
-    // Check if the warmup_fraction is done, we only consider the last level
-    // cache for warmup.
-    if (knob_warmup_fraction > 0.0 &&
-        llcache->get_loaded_fraction() > knob_warmup_fraction) {
+    // reset cache stats when warming up is completed
+    if (!is_warmed_up && check_warmed_up()) {
         for (int i = 0; i < knob_num_cores; i++) {
             icaches[i]->get_stats()->reset();
             dcaches[i]->get_stats()->reset();
         }
         llcache->get_stats()->reset();
+        is_warmed_up = true;
```

Comment:
This seems better set inside check_warmed_up(): otherwise every caller has to set it.